pkg/util/password/dispatch.go

```diff
@@ -43,6 +51,10 @@ func Compare(password, hash []byte) error {
 }
 
 func TryMigrate(password []byte, hash *[]byte) (migrated bool, err error) {
+	if len(password) > MaxLength {
+		password = password[:MaxLength]
+	}
+
```

Comment:
We should only limit the length of newly proposed password. For existing password, we do not place any limits on it. This allows old, long password continue to work.